File: Archiver/frmLoadTest.vb
Imports System.IO

Public Class frmLoadTest
    Private Sub lbFiles_SelectedIndexChanged(sender As Object, e As EventArgs) Handles lbFiles.SelectedIndexChanged

    End Sub

    Private Sub frmLoadTest_Load(sender As Object, e As EventArgs) Handles MyBase.Load

    End Sub

    Private Sub btnDir_Click(sender As Object, e As EventArgs) Handles btnDir.Click

        Dim result As DialogResult = FolderBrowserDialog1.ShowDialog()

        ' Test result.
        If result = Windows.Forms.DialogResult.OK Then
            txtDir.Text = FolderBrowserDialog1.SelectedPath
        End If

    End Sub

    Private Sub btnRun_Click(sender As Object, e As EventArgs) Handles btnRun.Click
        Dim dir As New DirectoryInfo(txtDir.Text)

        lbDirs.Items.Clear()
        lbDirs.Items.Add(dir)
        lbDirs.Sorted = True

        For Each dirItem As DirectoryInfo In dir.GetDirectories
            lbDirs.Items.Add(dirItem.FullName)
        Next

    End Sub

    Private Sub lbDirs_SelectedIndexChanged(sender As Object, e As EventArgs) Handles lbDirs.SelectedIndexChanged
        Dim targetDirectory As String = ""
        targetDirectory = lbDirs.SelectedItems(0).ToString
        Dim txtFilesArray As String() = Nothing
        If ckSubdir.Checked Then
            txtFilesArray = Directory.GetFiles(targetDirectory, "*.*", SearchOption.AllDirectories)
        Else
            txtFilesArray = Directory.GetFiles(targetDirectory, "*.*", SearchOption.TopDirectoryOnly)
        End If

        lbFiles.Items.Clear()
        lbFiles.Sorted = True

        For Each fname As String In txtFilesArray
            lbFiles.Items.Add(fname)
        Next
    End Sub

    Private Sub Button1_Click(sender As Object, e As EventArgs) Handles Button1.Click
        Dim SearchTgt As String = txtSearch.Text.ToLower
        Dim S As String = ""
        Dim J As Integer = 0
        Dim iCnt As Integer = 0
        lbFiles.SelectedItems.Clear()
        For I As Integer = 0 To lbFiles.Items.Count - 1
            S = lbFiles.Items(I).ToString.ToLower
            J = S.IndexOf(SearchTgt)
            If J >= 0 Then
                lbFiles.SetSelected(I, True)
                iCnt += 1
            End If
        Next
        MessageBox.Show("Items Found: " + iCnt.ToString)
    End Sub

End Class

File: Archiver/frmLoadTest.cs
using System;
using global::System.IO;
using System.Windows.Forms;
using MODI;

namespace EcmArchiver
{
    public partial class frmLoadTest
    {
        public frmLoadTest()
        {
            InitializeComponent();
            _lbFiles.Name = "lbFiles";
            _lbDirs.Name = "lbDirs";
            _btnRun.Name = "btnRun";
            _btnDir.Name = "btnDir";
            _Button1.Name = "Button1";
        }

        private void lbFiles_SelectedIndexChanged(object sender, EventArgs e)
        {
        }

        private void frmLoadTest_Load(object sender, EventArgs e)
        {
        }

        private void btnDir_Click(object sender, EventArgs e)
        {
            var result = FolderBrowserDialog1.ShowDialog();

            // Test result.
            if (result == DialogResult.OK)
            {
                txtDir.Text = FolderBrowserDialog1.SelectedPath;
            }
        }

        private void btnRun_Click(object sender, EventArgs e)
        {
            var dir = new DirectoryInfo(txtDir.Text);
            lbDirs.Items.Clear();
            lbDirs.Items.Add(dir);
            lbDirs.Sorted = true;
            foreach (DirectoryInfo dirItem in dir.GetDirectories())
                lbDirs.Items.Add(dirItem.FullName);
        }

        private void lbDirs_SelectedIndexChanged(object sender, EventArgs e)
        {
            string targetDirectory = "";
            targetDirectory = lbDirs.SelectedItems[0].ToString();
            string[] txtFilesArray = null;
            if (ckSubdir.Checked)
            {
                txtFilesArray = Directory.GetFiles(targetDirectory, "*.*", SearchOption.AllDirectories);
            }
            else
            {
                txtFilesArray = Directory.GetFiles(targetDirectory, "*.*", SearchOption.TopDirectoryOnly);
            }

            lbFiles.Items.Clear();
            lbFiles.Sorted = true;
            foreach (string fname in txtFilesArray)
                lbFiles.Items.Add(fname);
        }

        private void Button1_Click(object sender, EventArgs e)
        {
            string SearchTgt = txtSearch.Text.ToLower();
            string S = "";
            int J = 0;
            int iCnt = 0;
            lbFiles.SelectedItems.Clear();
            for (int I = 0, loopTo = lbFiles.Items.Count - 1; I <= loopTo; I++)
            {
                S = lbFiles.Items[I].ToString().ToLower();
                J = S.IndexOf(SearchTgt);
                if (J >= 0)
                {
                    lbFiles.SetSelected(I, true);
                    iCnt += 1;
                }
            }

            MessageBox.Show("Items Found: " + iCnt.ToString());
        }
    }
}